src/pybind/mgr/volumes/pathspec.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+class SubvolumeSpec(object):
+    """
+    Specification of a subvolume, identified by (subvolume-id, group-id) tuple. Add fields as
+    required...
+    """
+
+    # where shall we (by default) create subvolumes
+    DEFAULT_SUBVOL_PREFIX = "/volumes"
+    # and the default namespace
+    DEFAULT_NS_PREFIX = "fsvolumens_"
+
+    # Reserved subvolume group name which we use in paths for subvolumes
+    # that are not assigned to a group (i.e. created with group=None)
+    NO_GROUP_NAME = "_nogroup"
+
+    def __init__(self, subvolumeid, groupid, subvolume_prefix=None, pool_ns_prefix=None):
+        assert groupid != SubvolumeSpec.NO_GROUP_NAME
+
+        self.subvolumeid = subvolumeid
+        self.groupid = groupid if groupid is not None else SubvolumeSpec.NO_GROUP_NAME
+        self.subvolume_prefix = subvolume_prefix if subvolume_prefix else SubvolumeSpec.DEFAULT_SUBVOL_PREFIX
+        self.pool_ns_prefix = pool_ns_prefix if pool_ns_prefix else SubvolumeSpec.DEFAULT_NS_PREFIX
+
+    def is_default_group(self):
+        """
+        Is the group the default group?
+        """
+        return self.groupid == SubvolumeSpec.NO_GROUP_NAME
+
+    @property
+    def subvolume_id(self):
+        """
+        Return the subvolume-id from the subvolume specification
```

Comment:
Do we need these comments?

Reply:
I do find it a tad useful although it's pretty much what we just return from the routine.